engine/gamesys/src/gamesys/scripts/script_image.cpp

```diff
@@ -102,8 +130,29 @@ namespace dmScript
         const char* buffer = lua_tolstring(L, 1, &buffer_len);
 
         bool premult = false;
-        if (top == 2) {
-            premult = lua_toboolean(L, 2);
+        bool buffer_output = false;
+
+        if (top >= 2)
+        {
+            if (lua_istable(L, 2))
+            {
+                lua_pushvalue(L, 2);
+
+                lua_getfield(L, -1, "premultiply_alpha");
+                premult = lua_isnil(L, -1) ? false : lua_toboolean(L, -1);
+                lua_pop(L, 1);
+
+                lua_getfield(L, -1, "buffer");
+                buffer_output = lua_isnil(L, -1) ? false : lua_toboolean(L, -1);
+                lua_pop(L, 1);
+
+                lua_pop(L, 1);
+            }
+            else
+            {
+                // DEPRECATED
+                premult = lua_toboolean(L, 2);
```

Comment:
Ideally, we should find these places in the code and log a dmLogWarningOnce message about deprecation and the suggested solution. MIght help with migration later on when we want to finally remove it.